frontend/views/containers/contributions/IncomeDetails.vue

```diff
@@ -0,0 +1,262 @@
+              type='radio'
+              name='incomeDetailsKey'
+              value='incomeAmount'
+              v-model='$v.form.incomeDetailsKey.$model'
+              @change='resetAmount'
+            )
+            i18n No, I don't
+
+          span.error(v-if='$v.form.incomeDetailsKey.$error')
+            i18n {{infoRequired}}
+
+        transition(name='expand')
+          fieldset(v-if='!!form.incomeDetailsKey')
+            label.field
+              .label {{ canPledge ? L('How much do you want to pledge?') : L('What\'s your monthly income?') }}
+              .input-combo(:class='{"error": $v.form.amount.$error }')
+                input.input(type='number' v-model='$v.form.amount.$model')
+                .suffix [$ USD]
+              span.error(v-if='$v.form.amount.$error') {{ inputAmountError }}
+              .helper(v-if='canPledge')
+                | Define up to how much you pledge to contribute to the group each month. Only the minimum needed amount will be given.
+              .helper(v-else)
+                text-who(:who='["[Rick]", "[Carl]", "[Kim]"]')
+                | &nbsp
+                i18n will ensure you meet the mincome
+            payment-methods(selected='manual')
+        .buttons
+          i18n.is-outlined(tag='button' @click='closeModal') Cancel
+          i18n(tag='button' type='submit' :disabled='$v.form.$invalid') Save
+        .c-feedback.has-text-1(
+          v-if='ephemeral.formFeedbackMsg.text'
+          :class='ephemeral.formFeedbackMsg.class'
+        ) {{ephemeral.formFeedbackMsg.text}}
+      .c-graph
+        | WIP: A graph is on its way!
+        //
+          group-pledges-graph.c-graph(
+            :userpledgeamount='canPledge && !!form.pledge ? Number(form.pledge) : null'
+            :userincomeamount='needsIncome && !!form.income ? Number(form.income) : null'
+          )
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import { validationMixin } from 'vuelidate'
+import { required } from 'vuelidate/lib/validators'
+import sbp from '~/shared/sbp.js'
+import InputAmount from './InputAmount.vue'
+import PaymentMethods from './PaymentMethods.vue'
+import Tooltip from '@components/Tooltip.vue'
+import ModalBaseTemplate from '@components/Modal/ModalBaseTemplate.vue'
+import TextWho from '@components/TextWho.vue'
+import GroupPledgesGraph from '../GroupPledgesGraph.vue'
+import currencies from '@view-utils/currencies.js' // TODO make filter
+import L from '@view-utils/translations.js'
+
+export default {
+  name: 'IncomeDetails',
+  mixins: [validationMixin],
+  components: {
+    ModalBaseTemplate,
+    InputAmount,
+    TextWho,
+    Tooltip,
+    PaymentMethods,
+    GroupPledgesGraph
+  },
+  data () {
+    return {
+      ephemeral: {
+        formFeedbackMsg: {}
+      },
+      form: {
+        incomeDetailsKey: null,
+        amount: null
+      },
+      fakeStore: {
+        incomeDetailsKey: 'pledgeAmount', // incomeAmount || pledgeAmount
+        amount: 100
```

Comment:
You should use the actual data now instead of a fakeStore since it's been implemented (see Contributions.vue, you can copy/paste)

Reply:
you started the review too soon, its done now :P

Reply:
? The `fakeStore` is still here